### ISSUE TITLE
Potential name clash between nested types and fields in schema components

### ISSUE THREAD

**Issue by jamiebrynes7** [spatialos/gdk-for-unity#958]
**Affects:** Release v0.1.0 and up
**Internal Ticket:** [UTY-1416](https://improbableio.atlassian.net/browse/UTY-1416)

---

#### Description

Given a schema file like the following: 

```
type Foo {
    enum Bar {
        YES= 0;
        NO = 1;
        MAYBE = 2;
    }

    Bar bar = 1;
}
```

The code generator will generate invalid C# code that does not compile.

> Note this occurs when the name of a nested type and the name of an field inside a component are _the same_.

#### Workaround

Rename either the nested type or the field.

**Comment by paulbalaji** [comment]
A resolution has been merged and will be part of our 0.3.1 release next week.

**Comment by jamiebrynes7** [comment]
Released in https://github.com/spatialos/gdk-for-unity/releases/tag/0.3.1
